### ISSUE TITLE
How is MSLE calculated and the training process of the model?

### ISSUE THREAD

**Issue by Phe2703** [MrYansong/CasSeqGCN#1]
1. The calculation formula of MSLE in the paper should take the logarithm of the predicted value and the actual value, but the code does not take the logarithm of the predicted value. Is this correct?
2. It seems that the code does not specify the use of GPU. Did the author complete the model training on the CPU? And is it actually still traversing for each batch?
3. Also, after each evaluation on the validation set, the evaluation on the test set is performed, what is the reason for this? Are the experimental results given in the paper taken from the last evaluation?

**Comment by MrYansong** [comment]
I am very glad that you are interested in our work.
1. The calculation method of MSLE is similar to that of DeepCas, CasCN, etc. They calculate the logarithm of the actual value but don't take the logarithm of the predicted value. I tried using the logarithm of the predicted value and it didn't affect the result.
2. I tried using the GPU, but gave it up due to some configuration issues. It is traversing for each batch, it can be optimized here.
3. The validation set is initially used as a condition for the model to stop running. To verify the convergence of the model, we set a fixed epoch and used the last test value as the experimental result.
